### PR TITLE
Change default values of ``ignore-imports`` and ``ignore-signatures``

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :hammer: Refactoring   |

## Description

See https://github.com/PyCQA/pylint/pull/6121/files#r844920300

### CODE REVIEW

**Comment by DanielNoord** [comment]
Blocked by #6121.

**Comment by DanielNoord** [comment]
Blocked by #6121.

File: ChangeLog [hunk 1]
```diff
@@ -25,6 +25,8 @@ Release date: TBA
 
   Closes #4301
 
+* Updated the default values of ``ignore-imports`` and ``ignore-signatures`` to be ``True``.
```
Comment (Pierre-Sassoulas) [inline ChangeLog:28]:
```suggestion
* By default the similarity checker will now ignore imports and ignore function signatures when computing
 duplication. If you want to keep the previous behaviour set ``ignore-imports`` and ``ignore-signatures`` to ``False``.
```

File: doc/whatsnew/2.14.rst [hunk 2]
```diff
@@ -278,6 +278,8 @@ Other Changes
 
   Closes #5608
 
+* Updated the default values of ``ignore-imports`` and ``ignore-signatures`` to be ``True``.
+
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2.14.rst:282]:
```suggestion
```suggestion
* By default the similarity checker will now ignore imports and ignore function signatures when computing
 duplication. If you want to keep the previous behaviour set ``ignore-imports`` and ``ignore-signatures`` to ``False``.
```

```

**Review by Pierre-Sassoulas — COMMENTED** [review]
I suggested a "user friendlier" changelog, let me know what you think :)

File: doc/whatsnew/2.14.rst [hunk 3]
```diff
@@ -278,8 +278,8 @@ Other Changes
 
   Closes #5608
 
-* Updated the default values of ``ignore-imports`` and ``ignore-signatures`` to be ``True``.
-
+* By default the similarity checker will now ignore imports and ignore function signatures when computing
+ duplication. If you want to keep the previous behaviour set ``ignore-imports`` and ``ignore-signatures`` to ``False``.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2.14.rst:282]:
```suggestion
 duplication. If you want to keep the previous behaviour set ``ignore-imports`` and ``ignore-signatures`` to ``False``.
 
```

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by jacobtylerwalls** [comment]
Good change. Was there some thought given to updating the default for the output of `--generate-rcfile`?

**Comment by DanielNoord** [comment]
```ini
# Imports are removed from the similarity computation
ignore-imports=yes

# Signatures are removed from the similarity computation
ignore-signatures=yes
```

I get this on `main`? `--generate-rcfile` uses the default values unless other values are specified in the command line or configuration file.

**Comment by jacobtylerwalls** [comment]
Thanks for clarifying, I hadn't pulled from main. 😅

**Comment by DanielNoord** [comment]
> Related, what is the idea behind maintaining the `examples/pylintrc`? Okay to let that drift from the default one?

I think the idea was (once) to have that be up to date with ``--generate-rcfile`` so that we could deviate from default values in our own `pylintrc`. I would be fine with dropping it I guess.

**Pull request merged**